pkg/process/checks/pod.go

```diff
@@ -18,27 +18,32 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
 	k8sProcessors "github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors/k8s"
 	"github.com/DataDog/datadog-agent/pkg/orchestrator"
+	oconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
 	"github.com/DataDog/datadog-agent/pkg/process/config"
 	"github.com/DataDog/datadog-agent/pkg/process/util"
 	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/clustername"
 	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
 )
 
 // Pod is a singleton PodCheck.
-var Pod = &PodCheck{}
+var Pod = &PodCheck{
+	config: oconfig.NewDefaultOrchestratorConfig(),
+}
 
 // PodCheck is a check that returns container metadata and stats.
 type PodCheck struct {
 	sysInfo                 *model.SystemInfo
 	containerFailedLogLimit *util.LogLimit
 	processor               *processors.Processor
+	config                  *oconfig.OrchestratorConfig
 }
 
 // Init initializes a PodCheck instance.
-func (c *PodCheck) Init(cfg *config.AgentConfig, info *model.SystemInfo) {
+func (c *PodCheck) Init(_ *config.AgentConfig, info *model.SystemInfo) error {
 	c.containerFailedLogLimit = util.NewLogLimit(10, time.Minute*10)
 	c.processor = processors.NewProcessor(new(k8sProcessors.PodHandlers))
 	c.sysInfo = info
+	return c.config.Load()
```

Comment:
Just for my understanding, is this Load() idempotent? Now that it's called by the collector and the pod check, I wonder if it could cause any unexpected side-effect.

Reply:
Yes, it's idempotent, what worried me at first - that fetching the clustername is going to become slower, but it is in fact retrieved once and cached.

Reply:
Gotcha! Thanks for clarifying it!